### ISSUE TITLE
CLI/TUI?

### ISSUE THREAD

**Issue by 097115** [loki-project/session-desktop#1412]
Probably no chance for a command-line or TUI version? No such plans? :)

(Sorry for not exactly an issue but a general question...)

**Comment by Pascal-Paillier** [comment]
Seconding that! A CLI under linux would really be a nice addition here. Love what you guys are doing!

**Comment by beantaco** [comment]
Thirding this feature request. :-)

**Comment by beantaco** [comment]
I made a suggestion of adding support for command line options in #1503.

**Comment by argonatorUberTF** [comment]
Fourthing. A basic no frills text CLI with maybe vt52-vt100 terminal emulation like in the old Unix days.  The BBS never died. 
Maybe ANSI colors and extended text on monitors beyond 80x25 or 132 x50 ?
3270 terminal emulation for the geeks from IBM out there like me?

**Comment by DillFrescott** [comment]
This needs to happen!!

**Comment by BeholdersEye** [comment]
Any way to send Session messages programmatically would be very useful.

**Comment by n-stone** [comment]
Can't we repurpose parts of https://github.com/AsamK/signal-cli

**Comment by maltfield** [comment]
+1 but note that interacting with Session users programmatically isn't just for 1337 h4x0rs who want to talk with their friends on their linux laptops, it would be a huge boon for business adoption.

My use-case: I want to be able to send alerts from my server to my handset e2ee. Consider, for example, all of the useful bots used in Slack & Matrix. Having the ability to write a simple bash or python script to send messages over Session would be hugely useful.

**Comment by maltfield** [comment]
See also https://github.com/hesiod-project/node-session-client/

**Comment by VityaSchel** [comment]
> See also https://github.com/hesiod-project/node-session-client/

Hoping the developer will add support for onion routing and closed group chats

**Comment by VityaSchel** [comment]
I'm happy to announce that I'm building headless version of Session: https://github.com/VityaSchel/session-nodejs-bot

I spent a few days and with it you can already get latest conversations, read other user's names and right now I'm trying to debug sending messages

**Comment by maltfield** [comment]
@VityaSchel will it be possible to securely obtain it? npm is infamous for supply chain security issues.

**Comment by VityaSchel** [comment]
> @VityaSchel will it be possible to securely obtain it? npm is infamous for supply chain security issues.

Sure, just go ahead and clone my repo. Session-desktop already uses npm to download all packages. I will redistribute my project in npm in case someone wants to use it as module. All they need to do is just `import { initializeSession } from 'session-messenger-nodejs'` and they can start building bots with it

**Comment by maltfield** [comment]
I don't think we should consider any supply chat that requires the use of npm to be secure.
